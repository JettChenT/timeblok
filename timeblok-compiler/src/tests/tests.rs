use crate::{compile_deterministic, ir::ExactDateTime};
use insta::{assert_snapshot, glob};
use std::fs;

macro_rules! assert_compile_snapshot {
    ($input:expr, $base_time:expr) => {
        assert_snapshot!(compile_deterministic($input, $base_time.clone()).unwrap());
    };

    ($input:expr) => {
        assert_compile_snapshot!($input, ExactDateTime::from_ymd_hms(2023, 1, 1, 0, 0, 0));
    };
}


#[cfg(not(target_family = "wasm"))]
#[test]
fn run_tests(){
    glob!("bloks/*.tb", |path| {
        let input = fs::read_to_string(path).unwrap();
        assert_compile_snapshot!(&input);
    });
}

#[cfg(target_family = "wasm")]
#[test]
fn run_tests(){
    glob!("wsm-bloks/*.tb", |path| {
        let input = fs::read_to_string(path).unwrap();
        assert_compile_snapshot!(&input);
    });
}