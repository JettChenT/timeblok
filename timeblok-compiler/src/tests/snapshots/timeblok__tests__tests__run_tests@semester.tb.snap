---
source: timeblok-compiler/src/tests/tests.rs
expression: "compile_deterministic(&input,\n        ExactDateTime::from_ymd_hms(2023, 1, 1, 0, 0, 0).clone()).unwrap()"
input_file: timeblok-compiler/src/tests/bloks/semester.tb
---
BEGIN:VCALENDAR
VERSION:2.0
PRODID:ICALENDAR-RS
CALSCALE:GREGORIAN
BEGIN:VEVENT
DTEND:20230216T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230216T080000Z
SUMMARY:OI Club!
UID:c8e1eb22-a627-3daa-884c-2205cf78d075
END:VEVENT
BEGIN:VEVENT
DTEND:20230223T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230223T080000Z
SUMMARY:OI Club!
UID:9311744c-3746-3502-84c9-d06e8b5ea2d6
END:VEVENT
BEGIN:VEVENT
DTEND:20230302T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230302T080000Z
SUMMARY:OI Club!
UID:4bb9c1a0-ded7-3f16-93fb-bd2cbac9a815
END:VEVENT
BEGIN:VEVENT
DTEND:20230309T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230309T080000Z
SUMMARY:OI Club!
UID:78f8cf0b-0c6a-3fb8-9d24-1e4a8e882eaa
END:VEVENT
BEGIN:VEVENT
DTEND:20230316T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230316T080000Z
SUMMARY:OI Club!
UID:c8f0193b-cdc1-3f89-93db-2fc9698580bd
END:VEVENT
BEGIN:VEVENT
DTEND:20230323T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230323T080000Z
SUMMARY:OI Club!
UID:80c57eb0-2acb-36ac-89f9-95f0481c77a6
END:VEVENT
BEGIN:VEVENT
DTEND:20230330T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230330T080000Z
SUMMARY:OI Club!
UID:a04c4b4b-7d76-3479-84b0-646b305d32b4
END:VEVENT
BEGIN:VEVENT
DTEND:20230406T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230406T080000Z
SUMMARY:OI Club!
UID:f98be397-abfb-3f91-8448-29738a166724
END:VEVENT
BEGIN:VEVENT
DTEND:20230413T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230413T080000Z
SUMMARY:OI Club!
UID:ccfcea68-5ad0-3f02-9565-f4a496b398e6
END:VEVENT
BEGIN:VEVENT
DTEND:20230420T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230420T080000Z
SUMMARY:OI Club!
UID:e80160bc-f25a-3566-bf9b-a16e91ef6ee4
END:VEVENT
BEGIN:VEVENT
DTEND:20230427T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230427T080000Z
SUMMARY:OI Club!
UID:ad0ba92c-49be-3dd4-a57e-9937c835ee7a
END:VEVENT
BEGIN:VEVENT
DTEND:20230504T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230504T080000Z
SUMMARY:OI Club!
UID:021e4910-26aa-36f6-9b2e-40e7fa27f124
END:VEVENT
BEGIN:VEVENT
DTEND:20230511T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230511T080000Z
SUMMARY:OI Club!
UID:4c760b19-ee4a-34c2-bf58-4c2580f064d0
END:VEVENT
BEGIN:VEVENT
DTEND:20230518T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230518T080000Z
SUMMARY:OI Club!
UID:012a6243-4fe1-3b3f-9ef2-305b0f609f81
END:VEVENT
BEGIN:VEVENT
DTEND:20230525T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230525T080000Z
SUMMARY:OI Club!
UID:f55d153e-6007-32a5-894e-41217e3ff851
END:VEVENT
BEGIN:VEVENT
DTEND:20230601T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230601T080000Z
SUMMARY:OI Club!
UID:48221549-0cb2-3f55-9e57-ed1d230391bc
END:VEVENT
BEGIN:VEVENT
DTEND:20230608T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230608T080000Z
SUMMARY:OI Club!
UID:3d2e2f80-2f42-3dac-b41e-b4948b93d6b1
END:VEVENT
BEGIN:VEVENT
DTEND:20230615T090000Z
DTSTAMP:20221231T160000Z
DTSTART:20230615T080000Z
SUMMARY:OI Club!
UID:36273f37-0f38-3c24-a082-31e6542f612c
END:VEVENT
END:VCALENDAR

